### PR TITLE
New Windows image 1.5.0

### PR DESCRIPTION
**Pull request by matthew-rollitt**
New software 
Maya 2020
3ds-Max 2021
MtoA-4.1.1.1-2020 (Maya for Arnold)
MAXtoA-4.2.2.20-2021 (3ds-Max for Arnold)
vray_adv_50005_max2021_x64
vray_adv_50021_maya2020_x64
Removed all previous versions of 
Maya, 3ds-Max, Arnold and V-Ray

### CODE REVIEW

File: runner/Tests/TestConfiguration.json [hunk 1]
```diff
@@ -1,54 +1,5 @@
...
-        "parameters": "Tests/ffmpeg/create-animation-from-images/job.parameters.json",
-        "expectedOutput":"ffmpeg-animation-from-images.mp4"
-    },
-    {
-        "name": "imagemagick-resize-images",
-        "template": "../templates/imagemagick/resize-images/job.template.json",
-        "poolTemplate": "../templates/imagemagick/resize-images/pool.template.json",
-        "parameters": "Tests/imagemagick/resize-images/job.parameters.json",
-        "expectedOutput":"image0001.jpg"
-    },
```
Comment (davidkydd) [inline runner/Tests/TestConfiguration.json:23]: Why were these removed? Are they no longer present on the image?

File: runner/Tests/TestConfiguration.json [hunk 2]
```diff
@@ -63,149 +14,13 @@
...
-    {
-        "name": "blender-linux-dr",
-        "template": "../templates/blender/render-linux-dr/job.template.json",
-        "poolTemplate": "../templates/blender/render-linux-dr/pool.template.json",
-        "parameters": "Tests/blender/render-linux-dr/job.parameters.json",
-        "expectedOutput":"tile_001.png"
-    },
-    {
-        "name": "blender-windows-dr",
```
Comment (davidkydd) [inline runner/Tests/TestConfiguration.json:172]: as above - we are removing support for everything on blender except windows-
279?

File: runner/Tests/TestConfiguration.json [hunk 3]
```diff
@@ -214,143 +29,32 @@
...
-        "offer": "rendering-windows2016",
-        "version": "1.3.8"
+        "offer": "rendering-windows2016-preview",
+        "version": "1.5.0"
     },
     {
         "osType": "linux",
         "offer": "rendering-centos73",
-        "version": "1.1.7"
+        "version": "1.1.6"
```
Comment (davidkydd) [inline runner/Tests/TestConfiguration.json]: was it intentional to go from 1.1.7 to 1.1.6 for linux?

File: runner/Tests/maya/render-arnold-linux-container/job.parameters.2020.json [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
    "jobName":{
-      "value":"container-centos-maya2018-arnold"
+      "value":"container-centos-maya2020-arnold"
    },
    "poolId":{
-      "value":"container-centos-maya2018-arnold"
+      "value":"container-centos-maya2020-arnold"
    },
    "containerImage":{
       "value": "azurebatchrendering/centos-maya_2018u4_arnold2023:latest"
```
Comment (davidkydd) [inline runner/Tests/maya/render-arnold-linux-container/job.parameters.2020.json]: this will fail - need to remove support for it unless we are planning on pushing a new containerImage

File: runner/Tests/maya/render-vray-windows/job.parameters2020.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
 {
    "jobName":{
-      "value":"maya-vray-windows-2017"
+      "value":"maya-vray-windows-2020"
```
Comment (davidkydd) [inline runner/Tests/maya/render-vray-windows/job.parameters2020.json:3]: add corresponding test to testConfiguration.json

File: runner/Tests/maya/render-default-windows/job.parameters2020.json [hunk 6]
```diff
@@ -1,12 +1,12 @@
 {
    "jobName":{
-      "value":"maya-windows-default-2018"
+      "value":"maya-windows-default-2020"
```
Comment (davidkydd) [inline runner/Tests/maya/render-default-windows/job.parameters2020.json:3]: add corresponding test to testConfiguration.json

File: templates/arnold/render-windows-frames/job.template.json [hunk 7]
```diff
@@ -51,7 +51,7 @@
                 },
                 "repeatTask": {
                     "displayName": "frame {fileNameWithoutExtension}",
-                    "commandLine": "mkdir images & %ARNOLD_2018_EXEC% -l \"C:\\server\\mayaArnold\\2018\\shaders\" -i \"%AZ_BATCH_TASK_WORKING_DIR%\\assets\\{fileName}\" -dw -v 6 -o \"%AZ_BATCH_TASK_WORKING_DIR%\\images\\{fileNameWithoutExtension}.jpeg\"",
+                    "commandLine": "mkdir images & %ARNOLD_2020_EXEC% -l \"C:\\server\\mayaArnold\\2018\\shaders\" -i \"%AZ_BATCH_TASK_WORKING_DIR%\\assets\\{fileName}\" -dw -v 6 -o \"%AZ_BATCH_TASK_WORKING_DIR%\\images\\{fileNameWithoutExtension}.jpeg\"",
```
Comment (davidkydd) [inline templates/arnold/render-windows-frames/job.template.json]: \"C:\\server\\mayaArnold\\2018\\shaders\ is this directory correct for 2020?

File: templates/maya/render-arnold-linux/job.template.json [hunk 8]
```diff
@@ -20,15 +20,14 @@
         },
         "mayaVersion": {
             "type": "string",
-            "defaultValue": "Maya2019",
+            "defaultValue": "Maya2019",
```
Comment (davidkydd) [inline templates/maya/render-arnold-linux/job.template.json]: should this be 2020?

File: templates/maya/render-arnold-windows/job.template.json [hunk 9]
```diff
@@ -142,7 +134,7 @@
...
-                    "commandLine": "mkdir \"%AZ_BATCH_TASK_WORKING_DIR%\\thumbs\" & subst X: %AZ_BATCH_JOB_PREP_WORKING_DIR% && cmd.exe /v /c \"[variables('MayaVersions')[parameters('mayaVersion')].environmentValue] -renderer arnold -ai:lve 2 [parameters('additionalFlags')] -proj \"%AZ_BATCH_JOB_PREP_WORKING_DIR%\" -rd \"%AZ_BATCH_TASK_WORKING_DIR%\\images\" -s {0} -e {0} \"X:\\[parameters('sceneFile')]\"\" & subst X: /D",
+                    "commandLine": "mkdir \"%AZ_BATCH_TASK_WORKING_DIR%\\thumbs\" & set &  echo %AZ_BATCH_ACCOUNT_URL%>url.txt&echo %AZ_BATCH_SOFTWARE_ENTITLEMENT_TOKEN%>token.txt&powershell -command \"$characters=Get-ChildItem -Path Env:\\AZ_BATCH_ACCOUNT_URL;$token=Get-ChildItem -Path Env:\\AZ_BATCH_SOFTWARE_ENTITLEMENT_TOKEN;write-host 'url-length' $characters.value.length 'tokenlength' $token.value.length\"&subst X: %AZ_BATCH_JOB_PREP_WORKING_DIR% && cmd.exe /v /c \"[variables('MayaVersions')[parameters('mayaVersion')].environmentValue] -renderer arnold -ai:lve 2 [parameters('additionalFlags')] -proj \"%AZ_BATCH_JOB_PREP_WORKING_DIR%\" -rd \"%AZ_BATCH_TASK_WORKING_DIR%\\images\" -s {0} -e {0} \"X:\\[parameters('sceneFile')]\"\" & subst X: /D",
```
Comment (davidkydd) [inline templates/maya/render-arnold-windows/job.template.json]: this looks like logging for debugging purposes? Suggest removing

File: templates/maya/render-arnold-linux/job.template.json (hunk 8 above)
Comment (davidkydd) [reply templates/maya/render-arnold-linux/job.template.json]: recommend reverting all the changes to linux templates in this PR, and changing everything at once in a later PR as part of updating to a new linux image

File: templates/maya/render-vray-windows/job.template.json [hunk 10]
```diff
@@ -142,7 +134,7 @@
                             "elevationLevel": "admin"
                         }
                     },
-                    "commandLine": "mkdir \"%AZ_BATCH_TASK_WORKING_DIR%\\thumbs\" & subst X: %AZ_BATCH_JOB_PREP_WORKING_DIR% & cmd.exe /v /c \"[variables('MayaVersions')[parameters('mayaVersion')].environmentValue] -renderer vray [parameters('additionalFlags')] -proj \"%AZ_BATCH_JOB_PREP_WORKING_DIR%\" -rd \"%AZ_BATCH_TASK_WORKING_DIR%\\images\" -s {0} -e {0} \"X:\\[parameters('sceneFile')]\"\"",
+                    "commandLine": "set; mkdir \"%AZ_BATCH_TASK_WORKING_DIR%\\thumbs\" & set &subst X: %AZ_BATCH_JOB_PREP_WORKING_DIR% & cmd.exe /v /c \"[variables('MayaVersions')[parameters('mayaVersion')].environmentValue] -renderer vray [parameters('additionalFlags')] -proj \"%AZ_BATCH_JOB_PREP_WORKING_DIR%\" -rd \"%AZ_BATCH_TASK_WORKING_DIR%\\images\" -s {0} -e {0} \"X:\\[parameters('sceneFile')]\"\"",
```
Comment (davidkydd) [inline templates/maya/render-vray-windows/job.template.json]: double calls to "set" intentional?

**Review by davidkydd — CHANGES_REQUESTED** [review]
will review 3dsmax.ps1 separately as the diff is hard to visualize here

File: templates/3dsmax/scripts/3dsmax.ps1 [hunk 11]
```diff
@@ -80,39 +71,7 @@ function SetupDistributedRendering
...
-            $script:pre_render_script_content += "if index == 1 then (r.distributed_rendering = true)`r`n"            
-            $script:pre_render_script_content += "r.V_Ray_settings.system_vrayLog_level = 4`r`n"
-            $script:pre_render_script_content += "r.V_Ray_settings.system_vrayLog_file = ""$vrayLogFile""`r`n"
-        }    
-        ElseIf($renderer -eq "VRayAdv")
-        {
-            $script:pre_render_script_content += "index = findString rendererName ""V_Ray_Adv""`r`n"
-            $script:pre_render_script_content += "if index == 1 then (r.system_distributedRender = true)`r`n"
-            $script:pre_render_script_content += "r.system_vrayLog_level = 4`r`n"
-            $script:pre_render_script_content += "r.system_vrayLog_file = ""$vrayLogFile""`r`n"
```
Comment (davidkydd) [inline templates/3dsmax/scripts/3dsmax.ps1:112]: please add this line and the line above back in to the block for maxversion == 2021 and not VRayRT (currently line 136)

File: runner/Tests/TestConfiguration.json (hunk 1 above)
Comment (matthew-rollitt) [reply runner/Tests/TestConfiguration.json:23]: I'll add these 3 back in

File: runner/Tests/TestConfiguration.json (hunk 2 above)
Comment (matthew-rollitt) [reply runner/Tests/TestConfiguration.json:172]: added blender-dr back in

File: runner/Tests/TestConfiguration.json (hunk 3 above)
Comment (matthew-rollitt) [reply runner/Tests/TestConfiguration.json]: reverted back

File: runner/Tests/maya/render-vray-windows/job.parameters2020.json (hunk 5 above)
Comment (matthew-rollitt) [reply runner/Tests/maya/render-vray-windows/job.parameters2020.json:3]: yep that one is called "maya2020-vray-windows"

File: templates/arnold/render-windows-frames/job.template.json (hunk 7 above)
Comment (matthew-rollitt) [reply templates/arnold/render-windows-frames/job.template.json]: can be updated to 2020

File: templates/maya/render-vray-windows/job.template.json (hunk 10 above)
Comment (matthew-rollitt) [reply templates/maya/render-vray-windows/job.template.json]: removed

File: templates/maya/render-arnold-windows/job.template.json (hunk 9 above)
Comment (matthew-rollitt) [reply templates/maya/render-arnold-windows/job.template.json]: removed

File: templates/maya/render-arnold-linux/job.template.json (hunk 8 above)
Comment (matthew-rollitt) [reply templates/maya/render-arnold-linux/job.template.json]: reverted this change
